tests/runVmTests.js

```diff
@@ -81,53 +77,67 @@ function setupEnviroment (testData, rootVertex) {
   env.block.header.number = Buffer.from(testData.env.currentNumber.slice(2), 'hex')
   env.block.header.timestamp = Buffer.from(testData.env.currentTimestamp.slice(2), 'hex')
 
+  env.state = {}
   for (let address in testData.pre) {
-    const account = testData.pre[address]
-    const accountVertex = new Vertex()
-
-    accountVertex.set('code', new Vertex({
-      value: Buffer.from(account.code.slice(2), 'hex')
-    }))
-
-    accountVertex.set('balance', new Vertex({
-      value: Buffer.from(account.balance.slice(2), 'hex')
-    }))
-
-    for (let key in account.storage) {
-      accountVertex.set(['storage', ...Buffer.from(key.slice(2), 'hex')], new Vertex({
-        value: Buffer.from(account.storage[key].slice(2), 'hex')
-      }))
+    const testAccount = testData.pre[address]
+    let envAccount = {
+      code: testAccount.code,
+      balance: testAccount.balance,
+      storage: testAccount.storage
     }
 
-    const path = [...Buffer.from(address.slice(2), 'hex')]
-    rootVertex.set(path, accountVertex)
-    env.state = accountVertex
+    env.state[address] = envAccount
   }
 
   return env
 }
 
-async function checkResults (testData, t, instance, environment) {
-  // check gas used
-  t.equals(ethUtil.intToHex(environment.gasLeft), testData.gas, 'should have the correct gas')
+async function checkResults (testData, t, instance, environment, vmExceptionErr) {
+  // https://github.com/ethereum/tests/wiki/VM-Tests
+  // > It is generally expected that the test implementer will read env, exec 
+  // and pre then check their results against gas, logs, out, post and
+  // callcreates. If an exception is expected, then latter sections are absent
+  // in the test.
+
+  if ((testData.gas && testData.out && testData.post)) {
+    if (vmExceptionErr) {
+      t.fail('should not have VM exception')
+      console.log('vmExceptionErr:', vmExceptionErr)
```

Comment:
Is `log` emitted after `t.fail`?

Reply:
yea it logs.

Reply:
but should `t.fail` abort the execution?

Reply:
no, it's equivalent to doing `t.false(val, msg)` (assert that val is falsy), but i've already asserted that `vmExceptionErr` is falsy with `if (vmExceptionErr)`. I do it this way rather than `t.false(vmExceptionErr, 'should not have VM exception')` just so that there isn't an "ok  xxx should not have VM exception" for every test case.